ext/dom/element.c

```diff
@@ -874,8 +874,8 @@ PHP_METHOD(DOMElement, setAttributeNodeNS)
 	DOM_GET_OBJ(attrp, node, xmlAttrPtr, attrobj);
 
 	if (attrp->type != XML_ATTRIBUTE_NODE) {
-		php_error_docref(NULL, E_WARNING, "Attribute node is required");
-		RETURN_FALSE;
+		zend_argument_value_error(1, "must have node attribute");
```

Comment:
This message is not clear to me, maybe
```suggestion
		zend_argument_value_error(1, "must be an attribute node");
```
if I understood the error condition right? Also doesn't seem to be tested :(